merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -69,6 +69,48 @@ export class Constraint {
       expression: expression(new ActivityInstance(activityType, alias)).__astNode,
     });
   }
+
+  /**
+   * Detect when a spans object's cumulative duration either exceeds or falls short of a threshold within any interval of a given width.
+   *
+   * Violations can be reported in various ways by setting the `algorithm` argument:
+   * - `ExcessSpans` detects times when the duration exceeds the threshold and highlights the individual spans that
+   *    contributed to the threshold violation.
+   * - `ExcessHull` detects times when the duration exceeds the threshold and highlights the whole group of spans that
+   *    contributed to the threshold violation in one interval.
+   * - `DeficitSpans` detects times when the duration falls short of the threshold and highlights the individual gaps between spans
+   *    that contributed to the threshold violation.
+   * - `ExcessHull` detects times when the duration falls short of the threshold and highlights the whole group of gaps between
```

Comment:
Typo here and on L1111:
```suggestion
   * - `DeficitHull` detects times when the duration falls short of the threshold and highlights the whole group of gaps between
```

Reply:
Additionally can you remind me why this code needs to be written twice, once here and once in the `declare global` block?

Reply:
The first declaration (and implementation) is needed to get the class objects added to `globalThis` in javascript. This allows only the class values to be accessible without importing. `declare global` is needed to get the class types accessible in the constraint code without importing.

The doc comments are duplicated because monaco uses docs from the global declaration for mouse-over hints, and typedoc uses the normal implementation for generating docs.